File: board_dsl/src/main/java/com/dooho/board/entity/CommentEntity.java
package com.dooho.board.entity;

import com.dooho.board.dto.comment.CommentDto;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Entity(name = "Comment")
@Table(name = "Comment")
public class CommentEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer commentId;
    private Integer boardNumber;
    private String userEmail;
    private String commentUserProfile;
    private String commentUserNickname;
    private LocalDate commentWriteDate;
    private String commentContent;

    public CommentEntity(CommentDto dto){
        this.commentId = dto.getCommentId();
        this.boardNumber = dto.getBoardNumber();
        this.userEmail = dto.getUserEmail();
        this.commentUserProfile = dto.getCommentUserProfile();
        this.commentUserNickname = dto.getCommentUserNickname();
        this.commentWriteDate = dto.getCommentWriteDate();
        this.commentContent = dto.getCommentContent();

    }
}

File: inflearn_basic_DB/board_Kotlin/src/main/java/com/dooho/board/api/comment/CommentEntity.kt
package com.dooho.board.api.comment

import com.dooho.board.api.comment.CommentDto
import jakarta.persistence.*
import java.time.LocalDate

@Entity(name = "Comment")
@Table(name = "Comment")
data class CommentEntity(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var commentId: Int? = null,
    var boardNumber: Int? = null,
    var userEmail: String? = null,
    var commentUserProfile: String? = null,
    var commentUserNickname: String? = null,
    var commentWriteDate: LocalDate? = null,
    var commentContent: String? = null
) {
    constructor(dto: CommentDto?) : this(
        dto?.commentId,
        dto?.boardNumber,
        dto?.userEmail,
        dto?.commentUserProfile,
        dto?.commentUserNickname,
        dto?.commentWriteDate,
        dto?.commentContent
    )
}